### DISCUSSION TITLE
How to get SortDirection from DataGridColumn?

### DISCUSSION THREAD

**Discussion by well-coder** [AvaloniaUI/Avalonia#16216] [Q&A] [answered]
Hello!
are there any ways to get `SortDirection` from `DataGridColumn`? 

```
    private void HandleActivation(CompositeDisposable disposables)
    {
        _stopsDataGrid = this.FindControl<DataGrid>("StopsDataGrid")!;
        Observable.FromEventPattern<DataGridColumnEventArgs>(_stopsDataGrid, "Sorting")
            .Select(pattern => pattern.EventArgs.Column)
            .Subscribe(SaveSortedColumn)
            .DisposeWith(disposables);
    }
```

I did this and expected to get `SortDirection` from column variable. But it doesn't seem to exists
   ```
 private void SaveSortedColumn(DataGridColumn column)
    {
        _dataGridSortedColumn = column;
        _sortDirection= ?
    }
```
Can you advise some other ways to solve this problem?

**Comment by timunie** [comment]
doesn't DataGrid have an event for Sorting?

**Reply by well-coder** [reply]
It does. I converted this event to `observable` in the example above and subscribe to event arg's property "Column". `DataGridColumnEventArgs `has only two properties: `Handled `and `Column`. So. I expected to get `SortDirection `property there like in WPF: 
https://learn.microsoft.com/ru-ru/dotnet/api/system.windows.controls.datagridcolumn.sortdirection?view=windowsdesktop-8.0
The problem is following: after my view model updated, the sorting in column disappears. And i'd like to remember it last direction (before update) and then manually restore it.

**Reply by timunie** [reply, marked as answer]
Yeah looks like it is internal. You can use Reflection, but that's probably 😈 .

- there is an open feature request, you can upvote it: #2407
- You can also try if TreeDataGrid is better suited for you (has a steep learning curve)

**Reply by well-coder** [reply]
Okay, thank you for help!
